Exam Creation and Update

Toemmsche opened the pull request:
# Added
- Empty Card for new exams
- Dialog that contains form rows for all required fields to create or adjust an exam
- Cupertino bottom sheet that handles inputs for the form

# Changed 
- dateOfExam always required

Toemmsche committed:
[+] Add user interface and bloc for exam creation

Toemmsche committed:
Merge branch 'feature/exams-api-integration' into feature/new-exam

# Conflicts:
#	src/pubspec.yaml

Toemmsche committed:
Merge branch 'feature/exams-api-integration' into feature/new-exam

# Conflicts:
#	src/pubspec.yaml

Toemmsche committed:
[+] Upload exams on submission

Toemmsche committed:
[+] Add functionality to edit exams

tosoikea committed:
Update local.properties

tosoikea commented on src/lib/blocs/exams/exam_details_bloc.dart:
```diff
@@ -0,0 +1,104 @@
+    on<ExamSubmitted>(_onExamSubmitted);
+  }
+
+  _onNewExamOpened(NewExamOpened event, Emitter<ExamDetailsState> emit) {
```
Standardmäßig wäre diese Funktion vom Rückgabetyp "dynamic". Ist zwar nur "sugar", aber ein "void" als explizite Typangabe würde mir hier besser gefallen.

Toemmsche committed:
Merge branch 'dev' into feature/new-exam

Toemmsche committed:
[~] Add explicit void return type for event handleto

Toemmsche committed:
Merge remote-tracking branch 'origin/feature/new-exam' into feature/new-exam

Toemmsche committed:
[+] Add handling for invalid inputs
[~] Encapsulate UI redundancy in ExamFormRow
[+] Add isSameDate() extension method for DateTime

Toemmsche committed:
[-] Temporarily remove error text as it is not updating properly
